astartes/utils/convert_to_array.py

```diff
@@ -0,0 +1,30 @@
+import warnings
+
+import numpy as np
+
+from astartes.utils.exceptions import UncastableInputError
+from astartes.utils.warnings import ConversionWarning
+
+
+def convert_to_array(obj: object, name: str):
+    """Attempt to convert obj named name to a numpy array, with appropriate warnings and exceptions.
+
+    Args:
+        obj (object): The item to attempt to convert.
+        name (str): Human-readable name for printing.
+    """
+    warnings.warn(
+        "Attempting to cast {:s} to a numpy array, which may result in unexpected behavior "
+        "(remove this warning by passing numpy arrays directly to astartes).".format(
+            name,
+        ),
+        ConversionWarning,
+    )
+    try:
+        new_array = np.asarray(obj)
+    except Exception as e:
+        raise UncastableInputError(
+            "Unable to cast {:s} to a numpy array using np.asarray."
+        ) from e
+
```

Comment:
Could we add another try except block that checks if all values were cast to floats? Except for random splitting, I believe all splitting methods will require X to be an array of numbers so I think it could be helpful if we 1) try to convert "0" to 0 for example and 2) raise an error if the user accidentally passed actual strings as one of the columns in a data frame like "some description of each datapoint". Case 1 can commonly happen if reading data from a file and the user just forgets to type check the data. Case 2 could happen since not all columns are intended to be features but the user may forget to drop them beforehand.